### DISCUSSION TITLE
Can openid-client be used to validate OpenID Connect id tokens?

### DISCUSSION THREAD

**Discussion by dspinellis** [panva/openid-client#512] [Q&A] [answered]
If I understand the documented API correctly, it seems to be geared toward a web-based application workflow (hence e.g. the various endpoints that seem to be required when constructing an `Issuer` object). Can the library be used to simply validate an OpenID Connect id token obtained through a workflow independently of the openid-client library? It would be cool to have a generic solution for what is currently implemented by provider-specific libraries, such as those for [Google](https://developers.google.com/identity/sign-in/web/backend-auth#using-a-google-api-client-library) and [Apple](https://www.npmjs.com/package/verify-apple-id-token) tokens.

**Comment by mastermatt** [comment, marked as answer]
ID Tokens are required to be JSON Web Tokens (JWTs). This library uses `jose` (created by the same author) to verify JWTs and is probably what you're looking for.
https://github.com/panva/jose/blob/HEAD/docs/functions/jwt_verify.jwtVerify.md#readme
